File: Python/1_Creational Patterns/1_Singleton/Logger.py
class Logger:
    __instance = None

    @staticmethod
    def getInstance():
        if Logger.__instance == None:
            Logger()
        return Logger.__instance

    def __init__(self):
        if Logger.__instance != None:
            raise Exception("This class is a singleton!")
        else:
            Logger.__instance = self

    def log(self, message):
        # implementation of logging logic goes here
        print("Logging: " + message)


class MyApp:
    logger = Logger.getInstance()

    @staticmethod
    def main():
        MyApp.logger.log("Starting MyApp")

        # application logic goes here

        MyApp.logger.log("Shutting down MyApp")


if __name__ == "__main__":
    MyApp.main()

File: C#/1_Creational Patterns/1_Singleton/Logger.cs
public sealed class Logger {
    private static Logger instance = null;
    private static readonly object padlock = new object();

    Logger() {
        // private constructor to prevent instantiation
    }

    public static Logger Instance {
        get {
            lock (padlock) {
                if (instance == null) {
                    instance = new Logger();
                }
                return instance;
            }
        }
    }

    public void Log(string message) {
        // implementation of logging logic goes here
        Console.WriteLine("Logging: " + message);
    }
}

class MyApp {
    private static readonly Logger logger = Logger.Instance;

    static void Main(string[] args) {
        logger.Log("Starting MyApp");

        // application logic goes here

        logger.Log("Shutting down MyApp");
    }
}
